### DISCUSSION TITLE
Stable symbol / type?

### DISCUSSION THREAD

**Discussion by nipafx** [bonede/tree-sitter-ng#41] [Q&A] [answered]
Hi! First of all, thank you for this project - I struggled to find Java libraries with which I can highlight source code, but it's a breeze to build one with Tree-sitter. 💙🧡

To identify, e.g., keywords I need to know for each `TSNode` what it wraps. I started working off of `getSymbol` but I noticed that some very new (almost-)keywords like `yield` (symbol 84) intermingle with old ones, e.g. `return` (symbol 83). That made me think that these symbols aren't stable across Tree-sitter versions. Is that correct? If so, should I work off of `getType` or is there a better way to get that info?

**Comment by bonede** [comment, marked as answer]
It's better to use `TSNode#getType()` in my opinion. It's stable and and matches the name in the s-expression.

**Reply by nipafx** [reply]
Thanks for the answer, will do that. :)
